main.py

```diff
@@ -80,116 +75,157 @@ def __init__(self, profile_list, serial_port, servo, camera, RFID_reader):
-
+FOURCC = "*MJPG"
+CAMERA_INDEX = 0
+CAMERA_FPS = 60.0
+CAMERA_RES = (640,480)
 # RFID config
-serial_inerface_path = "/dev/ttyUSB0"
-baudrate = 9600
-RFID_proximity_BCM_pin_number = 23
+SERIAL_INTERFACE_PATH = "/dev/ttyUSB0"
+BAUDRATE = 9600
+RFID_PROXIMITY_BCM_PIN_NUMBER = 23
+# IR breaker config
+PHOTO_DIODE_BCM_PIN_NUMBER = 24
+# AnimalProfile config
+SESSION_SAVE_PATH = "./AnimalSessions/"
+VIDEO_SAVE_PATH = "./AnimalSessions/Videos/"
 
-# AnimalSession config
-session_save_path = "./AnimalSessions/"
 
 
+def main():
+   
 
+        main_logger.info("Creating AnimalProfiles...")
+	profile0 = AnimalProfile("0782B18367", "Jim Kirk", 0, 0, SESSION_SAVE_PATH, VIDEO_SAVE_PATH)
+	profile1 = AnimalProfile("0782B1797D", "Yuri Gagarin", 0, 0, SESSION_SAVE_PATH, VIDEO_SAVE_PATH)
+	profile2 = AnimalProfile("0782B191B5", "Elon Musk", 0, 0, SESSION_SAVE_PATH, VIDEO_SAVE_PATH)
+	profile3 = AnimalProfile("0782B19BCF", "Buzz Aldrin", 0, 0, SESSION_SAVE_PATH, VIDEO_SAVE_PATH)
+	profile_list = [profile0, profile1, profile2, profile3, profile4]
 
+        main_logger.info("Initializing servo")
+	servo_1 = servo.Servo(SERVO_PWM_BCM_PIN_NUMBER)
+        main_logger.info("Initializing camera")
+	camera_1 = camera.Camera(FOURCC, CAMERA_INDEX, CAMERA_FPS, CAMERA_RES)
+        main_logger.info("Initializing RFID reader")
+	RFID_1 = RFID.RFID_Reader(SERIAL_INTERFACE_PATH, BAUDRATE, RFID_PROXIMITY_BCM_PIN_NUMBER) 
+        main_logger.info("Initializing IR beam breaker")
+        IR_1 = IR.IRBeamBreaker(PHOTO_DIODE_BCM_PIN_NUMBER)
+        main_logger.info("Initializing SessionController")
+	session_controller = SessionController(profile_list, servo_1, camera_1, RFID_1, IR_1)
 
 
-def main():
-    
-	# Test animal profiles.
-	profile0 = AnimalProfile("0782B18622", "Jim Kirk", 0, 0, session_save_path)
-	profile1 = AnimalProfile("0782B182D6", "Yuri Gagarin", 0, 0, session_save_path)
-	profile2 = AnimalProfile("0782B17DE9", "Elon Musk", 0, 0, session_save_path)
-	profile3 = AnimalProfile("0782B18A1E", "Buzz Aldrin", 0, 0, session_save_path)
-	profile4 = AnimalProfile("5643564457", "Captain Picard", 0, 0, session_save_path)
-	profile_list = [profile0, profile1, profile2, profile3, profile4]
+	while True:
 
-        
-        # Initializing servo, camera and RFID reader and session controller.
-	servo_1 = servo.Servo(servo_PWM_BCM_pin_number)
-	camera_1 = camera.Camera(fourcc, camera_index, camera_fps, camera_res)
-	RFID_1 = RFID_Reader(serial_interface_path, baudrate, RFID_proximity_BCM_pin_number) 
-	session_controller = SessionController(profile_list, serial, servo_1, camera_1, RFID_1)
+            main_logger.info("Waiting for RF tag")
+	    print("Waiting for RF tag...")
+	    RFID_code = session_controller.RFID_reader.listenForRFID()
 
+            main_logger.info("Searching for AnimalProfile with matching RFID")
+            # Attempt to find a profile matching the detected RFID.
+	    profile = session_controller.searchForProfile(RFID_code)
```

Comment:
If you make the session controller throw an exception on a missing tag you can omit the -1 check:

```python
try:
    profile = session_controller.searchForProfile(RFID_code)
except:
   logging.warn('Unknown tag found')
   continue
````